### PR TITLE
WIP: welcome jdk17

### PR DESCRIPTION
**Pull request by brunolellis**
Motivation:
Provide JDK 17 on vertx starter tool.

WIP: it is still failing due to `kotin-maven-plugin` version 1.5.0 on vertx 3.9.10 as shown below:
```
[INFO] --- kotlin-maven-plugin:1.5.10:compile (compile) @ demo ---
[ERROR] Unknown JVM target version: 17
Supported versions: 1.6, 1.8, 9, 10, 11, 12, 13, 14, 15, 16
```

### CODE REVIEW

**Comment by tsegismont** [comment]
@brunolellis can you explain how to reproduce this failure?

File: build.gradle.kts [hunk 1]
```diff
@@ -18,6 +18,7 @@ import org.gradle.api.tasks.testing.logging.TestLogEvent.*
 
 plugins {
   id("io.vertx.vertx-plugin") version "1.2.0"
+  id("com.github.johnrengelman.shadow") version "7.1.0"
```
Comment (tsegismont) [inline build.gradle.kts:21]: Why is this needed?

**Comment by tsegismont** [comment]
I tried to reproduce on my laptop without success. This looks good to me. Thank you @brunolellis !

**Comment by brunolellis** [comment]
> @brunolellis can you explain how to reproduce this failure?

Just running the automated tests, may be some misconfiguration on my environment. 

Glad to hear it is working and I could help :D

File: build.gradle.kts (hunk 1 above)
Comment (tsegismont) [reply build.gradle.kts]: I guess it was because you built the project with jdk17

**Comment by tsegismont** [comment]
@brunolellis actually there was a build issue, it's fixed in #205

**Comment by brunolellis** [comment]
great!
